File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/inheritance/Role.java
package org.babyfish.jimmer.sql.model.inheritance;

import org.babyfish.jimmer.pojo.StaticType;
import org.babyfish.jimmer.sql.*;

@Entity
@StaticType(alias = "default", topLevelName = "TheRoleInput")
public interface Role extends RoleBase {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    long getId();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/inheritance/Role.kt
package org.babyfish.jimmer.sql.kt.model.inheritance

import org.babyfish.jimmer.sql.*

@Entity
interface Role : RoleBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Transient(RolePermissionCountResolver::class)
    val permissionCount: Int
}